src/sentry/auth/idpmigration.py

```diff
@@ -1,6 +1,28 @@
-import uuid
+from sentry import options
+from sentry.models import Organization, OrganizationMember, User
+from sentry.models.useremail import default_validation_hash
```

Comment:
Total nitpick, but having this function continue to live in `sentry.models.useremail` is slightly awkward if business logic depends in it in another place. (Granted, this is *almost* the same operation, but doesn't strictly have anything to do with the `UserEmail` model.)

We could move it to a general utility module. Or, because `default_validation_hash` is a simple one-liner, I don't think it would be wrong to call `get_random_string` here rather than to import and reuse it. (Btw, the `CHARACTERS` constant in `sentry.models.useremail` isn't really needed either. It's equivalent to `string.ascii_letters + string.digits`.)